specification/protocol/generate_rest.yaml

```diff
@@ -0,0 +1,186 @@
+openapi: 3.1.0
+info:
+  title: Open Inference API for text generation
+  description: Open Inference API for text generation
+  version: 1.0.0
+components:
+  schemas:
+    GenerateRequest:
+      type: object
+      required: 
+        - text_input
+      properties:
+        text_input:
+          type: string
+        parameters:
+          $ref: '#/components/schemas/GenerateParameters'
+    GenerateParameters:
+        allOf:
+          $ref: '#/components/schemas/GenerateParameters'
+        type: object
+        additionalProperties: {}
+        properties:
+          temperature:
+            type: number
+            format: float
+            default: null
+            minimum: 0
+            description: What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.
+          top_p:
+            type: number
+            format: float
+            maximum: 1
+            minimum: 0
+            description: An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered.
+          max_tokens:
+            type: integer
+            format: int32
+            default: 20
+            minimum: 0
```

Comment:
min is at least 1
https://github.com/vllm-project/vllm/blob/main/vllm/sampling_params.py#L186

Reply:
512 as upper limit is good?

Reply:
resolved